src/main/java/service/impl/ParserImpl.java

```diff
@@ -0,0 +1,30 @@
+package service.impl;
+
+import model.TransactionDto;
+import service.Parser;
+import service.Validator;
+
+public class ParserImpl implements Parser {
+    private static final int INDEX_OF_OPERATION = 0;
+    private static final int INDEX_OF_PRODUCT = 1;
+    private static final int INDEX_OF_QUANTITY = 2;
+    private static final String SEPARATOR = ",";
+    private final Validator validator;
+
+    public ParserImpl(Validator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public TransactionDto parsLine(String line) {
+        int quantity;
+        if (validator.validate(line)) {
```

Comment:
```suggestion
        validator.validate(line)
```
remove `if` - validator either throws exception or returns true